online-shop/src/app/components/containers/products-list/products-list.component.ts

```diff
@@ -1,20 +1,38 @@
-import { Component, OnInit } from '@angular/core';
-import { Product } from 'src/app/modules/shared/types/products.types';
+import { Component, OnChanges, OnInit, SimpleChanges } from '@angular/core';
+import { Product } from 'src/app/modules/shared/types/product.types';
+import { User } from 'src/app/modules/shared/types/user.types';
+import { AuthService } from 'src/app/services/auth.service';
 import { ProductService } from 'src/app/services/products.service';
 
 @Component({
   selector: 'app-products-list',
   templateUrl: './products-list.component.html',
   styleUrls: ['./products-list.component.scss']
 })
-export class ProductsListComponent implements OnInit {
+export class ProductsListComponent implements OnInit{
   products: Product[] = [];
-  constructor(private service: ProductService) { }
-
+  user!:User;
+  constructor(private service: ProductService, private authenticationService:AuthService) { }
   ngOnInit(): void {
     this.getProducts();
+    this.authenticationService.getUserProfile().subscribe(u=>
```

Comment:
It's not mandatory yet, but usually, in a bigger project, you want the lifecycles methods to be clean.  You can move the initialization code to a private class method and just call the method on the ngOnInit method.